manager/registry.py

```diff
@@ -216,7 +228,19 @@ def to_form(self):
             if key == "references" and isinstance(value, dict):
                 lines = ""
                 for x, y in value.items():
-                    lines += f"{x}:: {y}\n"
+                    logging.warning(f'item - {x}:: {y}')
+                    if x == 'http://schema.org/downloadUrl':
+                        try:
+                            if isinstance(y, list):
+                                # downloadUrl is a list of objects defining label + url
+                                # break it up into multiple lines of download/<label>:: <url>
+                                for u in y:
+                                    lines += f"download/{u['label']}:: {u['url']}\n"
+                        except JSONDecodeError as ex:
+                            # downloadUrl is a single string
+                            lines += f"{x}:: {y}\n"
+                    else:
+                        lines += f"{x}:: {y}\n"
```

Comment:
Hm, what if in the except block, we force a transformation to the new pattern? Something like

```
lines += f"download/Data Download:: {y}"
```

That way, when a form is opened for an old record, it will auto-populate the old download link into the new format, and then get saved appropriately. We can have students go through all the records then an check the download links, which will fix them.

Does that make sense?

Reply:
Absolutely! We could  adjust that easily here and/or in the `except` block, if that's what you mean:
```suggestion
                        lines += f"download/Data Download:: {y}\n"
```

Just note that `label` here needs to be unique, so if there are multiple links using the old format we may not be able to reliably use `download/Data Download::` as the prefix 🤔